SpatialGDK/Source/SpatialGDK/Private/Tests/MockObject.h

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Utils/RPCContainer.h"
+
+#include "Core.h"
+#include "MockObject.generated.h"
+
+using namespace SpatialGDK;
+
+TArray<uint8> TypeToArray(ESchemaComponentType Type);
+ESchemaComponentType ArrayToType(const TArray<uint8>& Array);
+
+UCLASS()
+class SPATIALGDK_API UMockObject : public UObject
```

Comment:
I recommend being precise with the names of test objects. The collective name for these are Test Doubles and following the naming conventions for the right type of double can be highly instructive to for the reader. Take a look at https://martinfowler.com/bliki/TestDouble.html for a good explanation.

In this instance, I think you are actually using this object as a Dummy in some of your tests and a Spy in others. So I would at least rename to use the spy term, and possible consider creating a second object to use which is a pure Dummy (it wouldn't need the ProcessedRPCIndicies).

Reply:
In addition, preferring names where the test double is a prefix might make it easier to find the one you want in intellisense / autocomplete once there are a lot of doubles in the codebase. So UObjectDummy and UObjectSpy in this case.

Reply:
It will also make the folder structure of the tests match the production folder structure more closely, which is good practice.

Reply:
I agree, it will make the test code cleaner, I'll apply this suggestion.

Reply:
Updated correspondingly with USpyObject and UDummyObject. Let me know if you prefer UObjectSpy and UDummySpy instead. I've also copied the explanation of object types from the page you've mentioned to TestDefinitions.h, but not sure if that is allowed, so maybe will have to remove it.

Reply:
Yeah, sorry, I think I do prefer the double on the end. When they go into their own files it will make more sense (to me anyway!) on disk to see:
Foo/Bar.h
Foo/Lar.h
...
Tests/Foo/BarDummy.h
Tests/Foo/BarTests.cpp
Tests/Foo/BarSpy.h
Tests/Foo/LarDummy.h
Tests/Foo/LarTests.cpp

Reply:
Moved and renamed the files. Now all RPCContainerTest files are located in
SpatialGDK\Source\SpatialGDKTests\Private\RPCContainer.
And the names for mock objects are ObjectDummy and ObjectSpy correspondingly.